### PR TITLE
Route around barriers

### PR DESCRIPTION
**Pull request by nCastle1**
This sample demonstrates how to calculate a route that hits several stops, without crossing any barriers. You can specify whether to find the best sequence, or to route to the sequence the user gave. 

Note: there are more files than usual because images needed to be added to each platform.

### CODE REVIEW

File: src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Network Analysis/RouteAroundBarriers/RouteAroundBarriers.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            // Guard against error conditions.
+            if (_routeParameters == null)
+            {
+                ShowMessage("Not ready yet", "Sample isn't ready yet; define route parameters first.");
+                return;
+            }
+
+            if (_stopsOverlay.Graphics.Count < 2)
+            {
+                ShowMessage("Not enough stops", "Add at least two stops before trying to route.");
```
Comment (ThadT) [inline src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Network Analysis/RouteAroundBarriers/RouteAroundBarriers.xaml.cs]: I think "trying to route" sounds a little awkward. How about "solving a route"?

File: src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Route around barriers
+
+Find a route that reaches all stops without crossing any barriers.
+
+![Screenshot](RouteAroundBarriers.jpg)
+
+## Use case
+
+You can define barriers to avoid unsafe areas, for example flooded roads, when planning the most efficient route to evacuate a hurricane zone. You could also use this functionality to plan routes when you know an area will be inaccessible due to a community activity like a run or a market night.
```
Comment (ThadT) [inline src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md]: Maybe a little more context?
"When solving a route, barriers allow you to define portions of the network that cannot be traversed. ..."

File: src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## How to use the sample
+
+Click 'Add stop' to add stops to the route. Click 'Add barrier' to add areas that can't be crossed by the route. Click 'Route' to find the route and display it.
+
+## How it works
+
+1. Create the route task by calling `RouteTask.CreateAsync(_serviceUrl)`.
+2. Get the default route parameters for the service by calling `_routeTask.CreateDefaultParametersAsync`.
+3. When the user adds a stop, add it to the route parameters.
+    1. Normalize the geometry; otherwise the route job would fail if the user included any stops over the international date line.
```
Comment (ThadT) [inline src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md]: More accurately, the 180th degree meridian. The International Date Line doesn't completely follow this meridian (also called the "anti-meridian").

File: src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    2. Get the name of the stop by counting the existing stops - `_stepsOverlay.Graphics.Count + 1`.
+    3. Create a composite symbol for the stop, combining a pushpin marker and a text symbol.
+    4. Create the graphic from the geometry and the symbol.
+    5. Add the graphic to the stops graphics overlay
```
Comment (ThadT) [inline src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md]: Missing a period (".").

File: src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+4. When the user adds a barrier, create a polygon barrier and add it to the route parameters.
+    1. Normalize the geometry (see **3a** above).
```
Comment (ThadT) [inline src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md]: The bullet you refer to is rendered in the markdown as "3i", not "3a".

File: src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    2. Buffer the geometry to create a larger barrier from the tapped point by calling `GeometryEngine.BufferGeodetic(mapLocation, 500, LinearUnits.Meters)`
+    3. Create the graphic from the geometry and the symbol.
+    4. Add the graphic to the barriers overlay.
+5. When ready to find the route, configure the route parameters.
+    1. Set the `ReturnStops` and `ReturnDirections` to true
```
Comment (ThadT) [inline src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md]: Missing period.

File: src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    2. Create a `Stop` for each graphic in the stops graphics overlay. Add that stop to a list, then call `_routeParameters.SetStops(routeStops)`.
+    3. Create a `PolygonBarrier` for each graphic in the barriers graphics overlay. Add that barrier to a list, then call `_routeParameters.SetPolygonBarriers(routeBarriers)`.
+    4. If the user will accept routes with the stops in any order, set `FindBestSequence` to true to find the most optimal route.
```
Comment (ThadT) [inline src/Android/Xamarin.Android/Samples/Network Analysis/RouteAroundBarriers/readme.md]: Should the values be formatted to make them a bit more readable? 
true => `true`

**Comment by nCastle1** [comment]
@saratchandrakarumuri I believe the Android and Forms Android samples should be much more reliable now. 

As for the XF UWP issue, I don't think I can fix that without risking the text not displaying at all on mobile. The way it is now ensures that it will be visible on all devices; I don't think it should be an issue to display in a different part of the map.

**Pull request merged**